tests/testsuite/build_script.rs

```diff
@@ -453,6 +453,43 @@ fn custom_build_env_var_rustc_linker() {
     p.cargo("build --target").arg(&target).run();
 }
 
+// Only run this test on linux, since it's difficult to construct
+// a case suitable for all platforms.
+// See:https://github.com/rust-lang/cargo/pull/12535#discussion_r1306618264
+#[cargo_test]
+#[cfg(target_os = "linux")]
+fn custom_build_env_var_rustc_linker_with_target_cfg() {
+    if cross_compile::disabled() {
+        return;
+    }
+
+    let target = cross_compile::alternate();
+    let p = project()
+        .file(
+            ".cargo/config",
+            r#"
+            [target.'cfg(target_arch = "i686")']
```

Comment:
My fault. `i686` is still x86 arch. Just the pointer is 32bits. Trust me one more time. This should work 🙇🏾.

```suggestion
            [target.'cfg(target_pointer_width = "32")']
```

Reply:
Thanks!